lib/easypost/shipment.rb

```diff
@@ -66,7 +66,7 @@ def lowest_rate(carriers=[], services=[])
 
       self.get_rates unless self.rates
 
-      carriers = carriers.is_a?(String) ? carriers.split(',') : Array(carriers)
+      carriers = carriers.is_a?(String) ? carriers.split(',') : Array(carriers).clone
```

Comment:
instead of `clone` -- it might be better to get rid of the `map!` -- as it could still cause problems down the road when this issue is forgotten.  we can extract this line into a method that does the cleanup:
```rb
carriers = normalized_list(carriers)

def normalized_list(list = []) # we can use a refinement here
  list = list.is_a?(String) ? list.split(',') : Array(list)
  list.map(&:downcase).map(&:strip)
end
```